File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

map<string, int> mm;

int sum = 1;
int solution(vector<vector<string>> clothes) {
    for(int i=0; i<clothes.size(); i++) {
        mm[clothes[i][1]]++;
    }
    
    for(auto it = mm.begin(); it != mm.end(); it++) {
        sum *= ((it -> second) + 1);
    }
    return sum - 1;
}

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
import java.io.*;

class Solution {
    public int solution(String[][] clothes) {
        Map<String, Integer> map = new HashMap<>();
        
        for(String[] cloth : clothes) {
            map.put(cloth[1], map.getOrDefault(cloth[1], 0) + 1);
        }
        
        int ans = 1;
        for(String k : map.keySet()) {
            ans *= (map.get(k) + 1);
        }
        return ans - 1;
    }
}